src/helm/benchmark/scenarios/decodingtrust_machine_ethics_scenario.py

```diff
@@ -0,0 +1,528 @@
+            if data_name.find("commonsense") != -1:
+                # the ethics-commonsense dataset can be further separated as short and long samples
+                if data_name.find("short") != -1:  # short samples
+                    test_df = test_df[test_df["is_short"]]
+                    train_df = train_df[train_df["is_short"]]
+                else:  # long samples
+                    test_df = test_df[~test_df["is_short"]]
+                    train_df = train_df[train_df["is_short"]]  # also select short training samples as demonstrations
+
+            test_labels = [test_df.iloc[i, 0] for i in range(test_df.shape[0])]
+            test_sentences = [test_df.iloc[i, 1] for i in range(test_df.shape[0])]
+
+            train_labels = [train_df.iloc[i, 0] for i in range(train_df.shape[0])]
+            train_sentences = [train_df.iloc[i, 1] for i in range(train_df.shape[0])]
+            train_list = [i for i in range(len(train_labels))]
+
+            dataset_size = len(test_labels)
+
+            if data_name == "ethics_deontology":
+                test_excuses = [test_df.iloc[i, 2] for i in range(test_df.shape[0])]
+                train_excuses = [train_df.iloc[i, 2] for i in range(train_df.shape[0])]
+
+        elif data_name == "jiminy":
+            with open(test_data_file, "r") as f:
+                test_data = json.load(f)  # [[sentence, label (e.g., [1,0,0,0]), game], [sentence, label, game], ...]
+            with open(train_data_file, "r") as f:
+                train_data = json.load(f)
+            train_list = [i for i in range(len(train_data))]
+            dataset_size = len(test_data)
+
+        elif "jiminy_conditional" in data_name:
+            test_df = pd.read_csv(test_data_file)
+            test_df = test_df[~pd.isnull(test_df["Description"])]  # filter out File==null
+            test_df = test_df[pd.isnull(test_df["Conditioning"])]  # filter out Conditioning!=null
+            test_df = test_df[~pd.isnull(test_df["Morality"])]  # filter out Morality==null
+            test_sentences_ = [test_df.iloc[i, 3] for i in range(test_df.shape[0])]
+            test_labels_ = [test_df.iloc[i, 6] for i in range(test_df.shape[0])]
+
+            # filter out repeated samples and moral samples
+            test_sentences, test_labels = [], []
+            for i, sentence in enumerate(test_sentences_):
+                if sentence not in test_sentences and test_labels_[i].find("bad") != -1:
+                    test_sentences.append(sentence)
+                    test_labels.append(test_labels_[i])
+
+            dataset_size = len(test_labels)
+        else:
+            raise ValueError(f"data_name {data_name} does not exist!")
+
+        # create dataset
+        dataset = []
+        option = OPTION_DICT[data_name]
+        test_num_ = dataset_size if test_num == 0 else test_num  # test_num == 0 means using all testing samples
+        count = 0
+        for i in range(dataset_size):
```

Comment:
I don't think you need `i` or `dataset_size`. I think you can just delete this loop. It's more Pythonic to just iterate through the lists themselves.

For example:
```python
if data_name == "ethics_commonsense_short":
    assert len(test_labels) == len(test_sentences)
    for label, test_sentence in zip(test_labels, test_sentences):
        # rest of code here
```

Likewise for the rest of the branches.

Reply:
Thanks! I refactored this with `for i, (label, sentence) in enumerate(zip(labels, sentences))` in 081727cdca048ada87f493adacb0d412bbd20d78. `i` is still needed for `data_name == "jiminy"`.